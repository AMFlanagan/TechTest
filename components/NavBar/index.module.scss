.navContainer {
  position: fixed;
  width: 100%;
  height: 56px;
  padding: 0;
  margin: 0 auto;
  background-color: #222222;
  box-shadow: 0px -1px 5px green;
  list-style-type: none;
}

.navBar {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 56px;
  margin: 0 auto;
  padding: 0;
}

.navItem {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 10px;
  padding: 0 10px;
  min-width: 80px;
  text-align: center;
}

.label {
  color: inherit;
  text-decoration: none;
}

.selected,
.navItem:hover {
  border-bottom: 2px solid green;
  color: green;
}

@media only screen and (min-width: 600px) {
  .navBar {
    max-width: 1100px;
  }
}
