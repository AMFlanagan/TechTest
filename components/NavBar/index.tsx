import React from 'react'
import Link from 'next/link'
// import formatText from '../utils/formatText'

import styles from './index.module.scss'
import { useRouter } from 'next/router'

interface IProps {
  eventKeys?: string[]
}

const NavBar: React.FC<IProps> = (props: IProps) => {
  const router = useRouter()

  return (
    <div className={styles.navContainer}>
      <ul className={styles.navBar}>
        <li
          className={`${styles.navItem} ${
            router.pathname === '/' ? styles.selected : ''
          }`}
        >
          <Link href="/">
            <a className={styles.label}>Home</a>
          </Link>
        </li>
        {/* {props.eventKeys?.map((eventKey: string, i: number) => {
        return (
          <li
            key={i}
            className={`${styles.navItem} ${
              router.pathname === `${eventKey}` ? styles.selected : ''
            }`}
          >
            <Link href={`/${eventKey}`}>
              <a className={styles.label}>{formatText(eventKey)}</a>
            </Link>
          </li>
        )
      })} */}
      </ul>
    </div>
  )
}

NavBar.displayName = 'NavBar'

export default NavBar
