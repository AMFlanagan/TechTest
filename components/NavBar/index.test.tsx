import { render, fireEvent } from '@testing-library/react'
import NavBar from '.'

const mockUseRouter = () => ({
  route: '',
  pathname: '',
  query: '',
  asPath: '',
  push: jest.fn(),
  events: {
    on: jest.fn(),
    off: jest.fn(),
  },
  beforePopState: jest.fn(() => console.log('skjdbfksjdfbdsjfbsd')),
  prefetch: jest.fn(() => null),
})

jest.mock('next/router', () => ({
  useRouter() {
    return mockUseRouter
  },
}))

describe('Event', () => {
  it('renders as expect', () => {
    const { getByText, asFragment } = render(<NavBar />)

    expect(getByText('Home')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders as expect on the homepage', () => {
    const { getByText, asFragment } = render(<NavBar />)

    expect(getByText('Home')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  describe('actions', () => {
    // TODO when more routes added
  })
})
