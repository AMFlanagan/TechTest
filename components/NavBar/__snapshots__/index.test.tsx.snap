// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders as expect 1`] = `
<DocumentFragment>
  <div
    class="navContainer"
  >
    <ul
      class="navBar"
    >
      <li
        class="navItem "
      >
        <a
          class="label"
          href="/"
        >
          Home
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Event renders as expect on the homepage 1`] = `
<DocumentFragment>
  <div
    class="navContainer"
  >
    <ul
      class="navBar"
    >
      <li
        class="navItem "
      >
        <a
          class="label"
          href="/"
        >
          Home
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
